scipio/src/timer.rs

```diff
@@ -422,8 +429,37 @@ impl<T: 'static> TimerAction<T> {
     /// ```
     /// [`TimerAction`]: struct.TimerAction
     pub async fn cancel(self) {
+        self.destroy();
+        self.join().await;
+    }
+
+    /// Cancel an existing [`TimerAction`], without waiting for it to return
+    ///
+    /// This is a non-async version of [`cancel`]. It will remove the timer if
+    /// it hasn't fired already and destroy the [`TimerAction`] releasing the resources
+    /// associated with it, but without blocking the current task. It is still possible
+    /// to [`join`] the task if needed.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use scipio::{LocalExecutor, TimerAction};
+    /// use std::time::Duration;
+    ///
+    /// LocalExecutor::spawn_executor("test", None, || async move {
+    ///     let action = TimerAction::once_in(Duration::from_millis(100), async move {
+    ///         println!("Will not execute this");
+    ///     });
+    ///     action.destroy();
+    ///     action.join().await;
+    /// });
+    /// ```
+    /// [`TimerAction`]: struct.TimerAction
+    /// [`cancel`]: struct.TimerAction.html#method.cancel
+    /// [`join`]: struct.TimerAction.html#method.join
+    pub fn destroy(&self) {
```

Comment:
The distinction between `destroy` and `cancel` seems subtle. What's the difference between calling `destroy()` vs calling `cancel().detach()`? Both are non-blocking.
Maybe you want to make the distinction clearer and why I'd want one over the other.

Reply:
The difference is indeed subtle. Behaviourally there is no difference so from a correctness point of view there is no difference.

From a performance point of view there is a difference:
`destroy()` does not generate a new task.
`cancel().detach()` does.

I thought I had mentioned that in the docs. What changes would you suggest to make that clearer ?

Reply:
if you don't want to wait `destroy` is always preferable. So maybe we just need to say that explicitly ?